### PR TITLE
[TACHYON-563] Update Javadoc in MasterClient

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses https://tachyon.atlassian.net/browse/TACHYON-563

### CODE REVIEW

File: common/src/main/java/tachyon/master/MasterClient.java [hunk 1]
```diff
@@ -112,9 +112,7 @@ public MasterClient(InetSocketAddress masterAddress, ExecutorService executorSer
    * @param length
    * @param checkpointPath
    * @return true if checkpoint is added for the <code>fileId</code> and false otherwise
-   * @throws FileDoesNotExistException
-   * @throws SuspectedFileSizeException
-   * @throws BlockInfoException
+   * @throws IOException
```
Comment (calvinjia) [inline common/src/main/java/tachyon/master/MasterClient.java]: Could you fill in the rest of the doc as well (ie. describe the parameters and conditions when exceptions will be thrown)?

Comment (jsimsa) [inline common/src/main/java/tachyon/master/MasterClient.java]: Done. PTAL

Comment (calvinjia) [inline common/src/main/java/tachyon/master/MasterClient.java]: Could you add the info for the exceptions after the throws clause? Ie. `@throws IOException if the file does not exist, has the wrong size, or its block information is corrupted.`

File: servers/src/main/java/tachyon/worker/block/BlockMasterSync.java [hunk 2]
```diff
@@ -109,9 +109,9 @@ public void registerWithMaster() throws IOException {
       mWorkerId =
           mMasterClient.worker_register(mWorkerAddress, storeMeta.getCapacityBytesOnTiers(),
               storeMeta.getUsedBytesOnTiers(), storeMeta.getBlockList());
-    } catch (BlockInfoException bie) {
-      LOG.error("Failed to register with master.", bie);
-      throw new IOException(bie);
+    } catch (IOException e) {
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/BlockMasterSync.java]: Could you change the variable name to `ioe`? (Applies to other parts of the code as well)

**Comment by calvinjia** [comment]
@jsimsa Thanks for the update, I left a few minor comments.

File: common/src/main/java/tachyon/master/MasterClient.java (hunk 1 above)
Comment (jsimsa) [inline common/src/main/java/tachyon/master/MasterClient.java]: Done.

File: servers/src/main/java/tachyon/worker/block/BlockMasterSync.java (hunk 2 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/BlockMasterSync.java]: Changes all occurrences of "catch (IOException e)" to "catch (IOException ioe)" in the three files modified by this PR.

File: common/src/main/java/tachyon/master/MasterClient.java [hunk 3]
```diff
@@ -383,10 +387,24 @@ public synchronized long getUsedBytes() throws IOException {
     return -1;
   }
 
+  /**
+   * Return a connection status.
+   *
+   * @return connection status
+   * @throws IOException
```
Comment (calvinjia) [inline common/src/main/java/tachyon/master/MasterClient.java]: This method doesn't seem to throw an exception?

File: common/src/main/java/tachyon/master/MasterClient.java [hunk 4]
```diff
@@ -404,6 +422,13 @@ public synchronized boolean isConnected() {
     return null;
   }
 
+  /**
+   * Check parameters.
+   *
+   * @param id to check
+   * @param path to check
+   * @throws IOException
```
Comment (calvinjia) [inline common/src/main/java/tachyon/master/MasterClient.java]: Could you add the exception conditions to this doc?

**Comment by calvinjia** [comment]
@jsimsa Looks good overall, left a few more minor comments.

File: common/src/main/java/tachyon/master/MasterClient.java (hunk 3 above)
Comment (jsimsa) [inline common/src/main/java/tachyon/master/MasterClient.java]: Done

File: common/src/main/java/tachyon/master/MasterClient.java (hunk 4 above)
Comment (jsimsa) [inline common/src/main/java/tachyon/master/MasterClient.java]: Done.

**Comment by calvinjia** [comment]
+1 LGTM

**Pull request merged**